Platform/CommonBoardPkg/Library/BoardSupportLib/BoardSupportLib.c

```diff
@@ -78,48 +79,50 @@ FillBootOptionListFromCfgData (
       break;
     }
 
-    if (BootOptionCfgData->ImageType == IMAGE_TYPE_NOT_USED) {
+    if ((BootOptionCfgData->ImageType == IMAGE_TYPE_NOT_USED) && (BootOptionCfgData->PreOsImageType == PREIMAGE_TYPE_NOT_USED)) {
       continue;
     }
 
-    UpdateFlag = 0;
     if (BootOptionCfgData->ImageType == IMAGE_TYPE_ADDENDUM) {
-      // This entry is an addendum for previous boot option entry
-      if (ImageIdx < ARRAY_SIZE(BootOption->Image)) {
-        UpdateFlag = 2;
-        BootOption = &OsBootOptionList->OsBootOption[PrevBootOptionIndex];
+      // This entry is an extra image addendum for previous boot option entry
+      BootOption = &OsBootOptionList->OsBootOption[PrevBootOptionIndex];
+      if (BootOptionCfgData->PreOsImageType > 0x3) {

```

Comment:
Use Enum type instead of 3

Reply:
0x3 is max extra image the boot option supported. so have a MACRO for it.